src/app/task/[id]/page.tsx

```diff
@@ -1,3 +1,173 @@
+"use client";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+import { useSelector } from "react-redux";
+import { useDispatch } from "react-redux";
+import {
+  selectTaskById,
+  selectCompletedSubtasksPercentage,
+} from "../../../redux/features/taskSlice";
+import {
+  handleCompleteSubtask,
+  handleDeleteTask,
+  repeatSubtasks,
+} from "../../../redux/features/taskSlice";
+import { RootState } from "../../../redux/store";
+import { BackIcon } from "../../icons/BackIcon";
+import { CalendarIcon } from "../../icons/CalendarIcon";
+import { PriorityIcon } from "../../icons/PriorityIcon";
+import { levelsCount } from "../../components/LevelSelector";
+import { ComplexityIcon } from "../../icons/ComplexityIcon";
+import { Progress } from "../../components/ui/progress";
+import { CompleteSubtaskIcon } from "../../icons/CompleteSubtaskIcon";
+import { RepeatIcon } from "../../icons/RepeatIcon";
+
+function determineLevel(value: number): string {
+  if (value < 4) {
+    return "Low";
+  } else if (value < 7) {
+    return "Medium";
+  } else {
+    return "High";
+  }
+}
+
 export default function TaskDetails({ params }: { params: { id: string } }) {
-  return <div>Task details: {params.id}</div>;
+  const dispatch = useDispatch();
+  const task = useSelector((state: RootState) =>
+    selectTaskById(state, params.id)
+  );
+
+  const completionPercentage = useSelector((state: RootState) =>
+    selectCompletedSubtasksPercentage(state, params.id)
+  );
+
+  const router = useRouter();
+
+  if (!task) {
+    return <div>No task found</div>;
+  }
+
+  return (
```

Comment:
The JSX of this file has so many className of tailwind css classes that it looks cluttered. You should create reusable component of those divs or span which uses same classes. For example you can create a component called Title which will have this css className="text-sm text-gray-600" which you can use for displaying Due Date and Priority title.